internal/provider/resource_vpc_router_route.go

```diff
@@ -0,0 +1,408 @@
+						Validators: []validator.String{
+							stringvalidator.OneOf(
+								string(oxide.RouteTargetTypeDrop),
+								string(oxide.RouteTargetTypeIp),
+								string(oxide.RouteTargetTypeInstance),
+								string(oxide.RouteTargetTypeInternetGateway),
+								string(oxide.RouteTargetTypeSubnet),
+								string(oxide.RouteTargetTypeVpc),
+							),
+						},
+					},
+					"value": schema.StringAttribute{
+						Description: "Depending on the type, it will be one of the following:" +
+							"- `vpc`: Name of the VPC " +
+							"- `subnet`: Name of the VPC subnet " +
+							"- `instance`: Name of the instance" +
+							"- `ip`: IP address " +
+							"- `internet_gateway`: Name of the internet gateway",
+						Optional: true,
+					},
+				},
+			},
+			"vpc_router_id": schema.StringAttribute{
+				Required:    true,
+				Description: "ID of the VPC router route that will contain the route.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Unique, immutable, system-controlled identifier of the VPC router route.",
+			},
+			"kind": schema.StringAttribute{
+				Computed:    true,
+				Description: "Whether the VPC router route is custom or system created.",
+			},
+			"time_created": schema.StringAttribute{
+				Computed:    true,
+				Description: "Timestamp of when this VPC router route was created.",
+			},
+			"time_modified": schema.StringAttribute{
+				Computed:    true,
+				Description: "Timestamp of when this VPC router route was last modified.",
+			},
+			"timeouts": timeouts.Attributes(ctx, timeouts.Opts{
+				Create: true,
+				Read:   true,
+				Update: true,
+				Delete: true,
+			}),
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *vpcRouterRouteResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan vpcRouterRouteResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createTimeout, diags := plan.Timeouts.Create(ctx, defaultTimeout())
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	ctx, cancel := context.WithTimeout(ctx, createTimeout)
+	defer cancel()
+
+	params := oxide.VpcRouterRouteCreateParams{
+		Router: oxide.NameOrId(plan.VPCRouterID.ValueString()),
+		Body: &oxide.RouterRouteCreate{
+			Description: plan.Description.ValueString(),
+			Destination: oxide.RouteDestination{
+				Type:  oxide.RouteDestinationType(plan.Destination.Type.ValueString()),
+				Value: plan.Destination.Value.ValueString(),
+			},
+			Name: oxide.Name(plan.Name.ValueString()),
+			Target: oxide.RouteTarget{
+				Type: oxide.RouteTargetType(plan.Target.Type.ValueString()),
+			},
+		},
+	}
+
+	if !plan.Target.Value.IsNull() {
+		params.Body.Target.Value = plan.Target.Value.ValueString()
+	}
```

Comment:
`ValueString` already returns an empty string if its value is null or unknown in the plan. We shouldn't need this check due to that behavior. Did you run into any behavior that led to creating this logic?

Reply:
Ah, no! I was trying out a different type in my attempt to deal with enums and forgot to remove that. Thanks!

Reply:
Actually no, checked it again. `Body.Target.Value` is type any, not string. So we don't want to set it as an empty string accidentally. I'll add a comment